packages/core/test/helper/reduce.js

```diff
@@ -13,10 +14,13 @@ import { newDefaultScheduler } from '@most/scheduler'
 * @param {function(result:*, x:*):*} f reducer function
 * @param {*} initial initial value
 * @param {Stream} stream to reduce
-* @returns {Promise} promise for the file result of the reduce
+* @returns {Promise} promise for the final result of the reduce
 */
-export const reduce = (f, initial, stream) =>
-  runEffects(new Reduce(f, initial, stream), newDefaultScheduler())
+export function reduce (f, initial, stream) {
+  let result = initial
+  const source = tap(x => { result = x }, new Reduce(f, initial, stream))
+  return runEffects(source, newDefaultScheduler()).then(() => result)
```

Comment:
I wonder if we should just ditch Reduce and ReduceSink entirely, and just use tap (since this is only used for testing).  E.g. maybe something like:

```js
export function reduce (f, initial, stream) {
  let result = initial
  const source = tap(x => { result = f(result, x) }, stream)
  return runEffects(source, newDefaultScheduler()).then(() => result)
}
```

and then just delete Reduce and ReduceSink?